node/libsignal_client.d.ts

```diff
@@ -8,9 +8,25 @@
 // FIXME: Eventually we should be able to autogenerate this.
```

Comment:
A broader discussion that we shouldn't have in a driveby PR comment:

TypeScript doesn't have a sound type system (succinctly described in [this blog post][0] if you're curious). That means that declaration files could contain lies, even if they're autogenerated. The soundest thing you can do is to declare most things as `unknown` and then use runtime type assertions. Onerous, but sound. (This [bit us at least once in RingRTC][1], though autogeneration would've fixed that.)

I think autogeneration is probably a reasonable middle ground, but I'd love to discuss our goals here. How important are sound types? Do we want to add runtime type assertions to make sure we're in the clear?

Again, probably not worth discussing here, but something I wanted to flag as it extends beyond this PR.

[0]: https://blog.logrocket.com/is-typescript-worth-it/
[1]: https://github.com/signalapp/ringrtc/commit/8c7d9ae0cba6dc596d86de2835c90bae312564e6#diff-d4fb1204a18470143330eacf6a818ddc5a550610e8e57409c102d44d253b03eaR1519

Reply:
The types in this file are not real JavaScript types; they're all going to be Node "[externals](https://nodejs.org/api/n-api.html#n_api_napi_create_external)". So lies are exactly what we want (to use TypeScript's type system to protect us). The autogeneration comment is mostly about avoiding writing all these functions out when it's already in the Rust files.